rustup-init.ps1

```diff
@@ -0,0 +1,142 @@
+    } elseif ($IsLinux) {
+        $ostype = "unknown-linux-gnu"
+    } else {
+        Write-Failure "unrecognized OS type"
+    }
+
+    if ([System.IntPtr]::Size -eq 8) {
+        $cputype = "x86_64"
+    } elseif ([System.IntPtr]::Size -eq 4) {
+        $cputype = "i686"
+    } else {
+        Write-Failure "unknown CPU type"
+    }
+
+    $arch = "${cputype}-${ostype}"
+    Write-Verbose "Detected arch '$arch'"
+    $arch
+}
+
+# If RUSTUP_UPDATE_ROOT is unset, default it.
+function Get-RustupUpdateRoot {
+    $EnvVar = [System.Environment]::GetEnvironmentVariable("RUSTUP_UPDATE_ROOT")
+    if (($EnvVar -ne $null) -and (Test-Path -Path $EnvVar)) {
+        $EnvVar
+    } else {
+        "https://static.rust-lang.org/rustup"
+    }
+}
+
+function Install-Rustup([AllowEmptyCollection()] [string[]]$Argv) {
+    $arch = Get-Architecture
+    switch -Wildcard ($arch) {
+        "*windows*" { $ext = ".exe"; break }
+        default { $ext = "" }
+    }
+
+    $url = "$(Get-RustupUpdateRoot)/dist/$arch/rustup-init$ext"
+
+    $dir = New-TemporaryDirectory
+    $file = Join-Path $dir "rustup-init$ext"
+
+    try {
+        if (($IsWindows -eq $null) -or $IsWindows) { Set-HighestEncryption }
+
+        Write-Verbose "Downloading $url to $file"
+        (New-Object System.Net.WebClient).DownloadFile($url, $file)
+        if ($IsMacOS -or $IsLinux) {
+          Write-Verbose "Setting execute permission on $file"
+          chmod u+x "$file"
+        }
+
+        Write-Verbose "Calling $file $Argv"
+        & "$file" @Argv
+    } finally {
+        Write-Verbose "Cleaning $dir"
+        Remove-Item "$dir" -Force -Recurse
+    }
+}
+
+function New-TemporaryDirectory {
+    $parent = [System.IO.Path]::GetTempPath()
+    [string]$name = [System.Guid]::NewGuid()
+    New-Item -ItemType Directory -Path (Join-Path $parent $name)
+}
+
+function Set-HighestEncryption {
+    # Implementation adapted from the Chocolatey installer script,
+    # see: https://chocolatey.org/install.ps1
+
+    # Attempt to set highest encryption available for SecurityProtocol.
+    # PowerShell will not set this by default (until maybe .NET 4.6.x). This
+    # will typically produce a message for PowerShell v2 (just an info message
+    # though)
+    $OldSPM = [System.Net.ServicePointManager]::SecurityProtocol
+    try {
+        # Set TLS 1.2 (3072). Use integers because the enumeration values for
+        # TLS 1.2 won't exist in .NET 4.0, even though they are addressable if
+        # .NET 4.5+ is installed (.NET 4.5 is an in-place upgrade).
+        [System.Net.ServicePointManager]::SecurityProtocol = 3072
+    } catch {
+        Write-Failure @'
+rustup: Unable to set PowerShell to use TLS 1.2 due to old .NET Framework
+installed. If you see underlying connection closed or trust errors, you may
+need to do one or more of the following: (1) upgrade to .NET Framework 4.5+ and
+PowerShell v3, (2) use the Download of rustup-init.exe method of install. See
```

Comment:
For (2) here, as you have made this script support multiple OS's, I think you should just say "(2) Download an alternative install method from https://rustup.rs/ ."

Reply:
Even better, done!